### PR TITLE
Feat: BuildRainer inital implementation.

### PR DESCRIPTION
**Pull request by Tailp**
Initital implementation of buildrainner :). Only listen to websocket at localhost:9080 and print out message.

### CODE REVIEW

**Comment by monperrus** [comment]
Rule: if you add some code, add some tests.

Could you add a test here?

**Comment by Tailp** [comment]
> Rule: if you add some code, add some tests.
> 
> Could you add a test here?

Resolved. Awating for review @gluckzhang @monperrus . :)

File: repairnator/build-rainer/src/test/java/fr/inria/repairnator/BuildRainerTest.java [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+public class BuildRainerTest {
+
+	@Test
+	/**
+	 * This test if BuildRainer can connect
+	 * and communicate with a ws server
+	 */
+	public void testWebSocket()
+    {
+        TestServer.getInstance().run();
```
Comment (monperrus) [inline repairnator/build-rainer/src/test/java/fr/inria/repairnator/BuildRainerTest.java]: This does not test that the BuildRainer has received the message.

File: repairnator/build-rainer/src/main/java/fr/inria/repairnator/BuildRainer.java [hunk 2]
```diff
@@ -1,9 +1,51 @@
...
+
+    @Override
+    public void onMessage( String message ) {
+        System.out.println( "received: " + message);
+    }
+
+    @Override
+    public void onOpen( ServerHandshake handshakedata ) {
+        System.out.println( "opened connection" );
+        send("Hi");
```
Comment (gluckzhang) [inline repairnator/build-rainer/src/main/java/fr/inria/repairnator/BuildRainer.java]: `BuildRainer` should receive messages from the Travis websocket, right? So this `send("Hi")` seems not necessary. The test case should also test whether `BuildRainer` is able to receive messages.

File: repairnator/build-rainer/src/main/java/fr/inria/repairnator/TestServer.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import java.nio.ByteBuffer;
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+
+import org.java_websocket.WebSocket;
+import org.java_websocket.handshake.ClientHandshake;
+import org.java_websocket.server.WebSocketServer;
+
+//Test server for BuildRainer
+public class TestServer extends WebSocketServer {
```
Comment (gluckzhang) [inline repairnator/build-rainer/src/main/java/fr/inria/repairnator/TestServer.java]: This `TestServer` is used as a stub for your test cases. I would suggest to move this file to `src/test` folder.

**Comment by Tailp** [comment]
@gluckzhang @monperrus Thanks a lot for the reviews and resolved. Awating for new comments :) .

**Comment by monperrus** [comment]
Thanks. Looking forward to the next step.

**Pull request merged**